web/packages/extension/src/common.ts

```diff
@@ -32,6 +34,35 @@ class CheckboxOption implements OptionElement<boolean> {
     }
 }
 
+class MaxExecutionDuration_NumberInputOption
+    implements OptionElement<Duration>
+{
+    constructor(
+        private readonly numberInput: HTMLInputElement,
+        readonly label: HTMLLabelElement
+    ) {}
+
+    get input() {
+        return this.numberInput;
+    }
+
+    get value() {
+        return Math.max(1, Math.round(parseInt(this.numberInput.value)));
+    }
+
+    set value(value: Duration) {
+        let newValue = 0;
+        if (isObsoleteDuration(value)) {
```

Comment:
Is this case even possible? I thought `value` is guaranteed to be a `number` here.

Reply:
Yeah, casting to `number` would be nicer IMO.

Reply:
I have removed `isObsoleteDuration`. I did not actually use a type-cast, now that I think of it, instead a `typeof`, which I would guess is better given that it is still type-safe given Typescript's flow typing.